### PR TITLE
Merge two RecordBatch

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/src/arrow.rs [hunk 1]
```diff
@@ -235,11 +235,6 @@ impl FixedSizeBinaryArrayExt for FixedSizeBinaryArray {
 
 /// Extends Arrow's [RecordBatch].
 pub trait RecordBatchExt {
-    /// Get a column by its name.
-    ///
-    /// Returns None if the column does not exist.
-    fn column_with_name(&self, name: &str) -> Option<&ArrayRef>;
```
Comment (eddyxu) [inline rust/src/arrow.rs:241]: Upstream `arrow-rs` provided a similar function recently. Let's remove our own implementation.

File: rust/src/index/vector/flat.rs [hunk 2]
```diff
@@ -76,19 +76,13 @@ impl VectorIndex for FlatIndex<'_> {
...
                 let k = params.key.clone();
                 let batch = batch.clone();
-                let vectors = batch
-                    .column_with_name(&self.column)
-                    .ok_or(Error::Schema(format!(
-                        "column {} does not exist in dataset",
-                        self.column,
-                    )))?
-                    .clone();
+                let vectors = batch[&self.column].clone();
```
Comment (changhiskhan) [inline rust/src/index/vector/flat.rs:79]: return err if it doesn't exist?

**Review by changhiskhan — APPROVED** [review]
pending the unwrap/error thing

File: rust/src/index/vector/flat.rs (hunk 2 above)
Comment (eddyxu) [reply rust/src/index/vector/flat.rs]: done.

**Pull request merged**